bin/git-forgit

```diff
@@ -137,10 +137,10 @@ _forgit_diff() {
     #   oldfile\0
     # We have to do a two-step sed -> tr pipe because OSX's sed implementation does
     # not support the null-character directly.
-    get_files="echo {} | sed 's/\\\\s*\\\\[.]\\\\s*//' | sed 's/  ->  /\\\n/' | tr '\\\n' '\\\0'"
+    get_files="echo {} | sed -e 's/^[[:space:]]*\\\\[[A-Z0-9]*\\\\][[:space:]]*//' | sed 's/.*->  //' | tr '\\\n' '\\\0'"
```

Comment:
This still does not work for me when diffing renamed files that were added. To reproduce:

1. mv LICENSE LICENSE1
2. git add .
3. gd --staged

The preview will look like LICENSE1 has been added. This can be fixed like this:

```suggestion
    get_files="echo {} | sed -e 's/^[[:space:]]*\\\\[[A-Z0-9]*\\\\][[:space:]]*//' | sed 's/.*->  //' | tr '\\\n' '\\\0'"
    get_files="echo {} | sed -e 's/^[[:space:]]*\\\\[[A-Z0-9]*\\\\][[:space:]]*//' | sed 's/  ->  /\\\n/' | tr '\\\n' '\\\0'"
```

The second sed command is what has changed here. I realized that I accidentally modified it with #354. Sorry you for that and another strong yes for unit tests.

Reply:
I think I see what you're seeing, but your suggestion doesn't work on my mac. In my current code I have on this branch, I get this:

<img width="1438" alt="Screenshot 2024-03-12 at 2 14 54 PM" src="https://github.com/wfxr/forgit/assets/14171107/1d33c672-78c9-483d-adbe-5f85504b53cd">


Your suggestion leads to a blank preview (on both ubuntu and mac):
<img width="1438" alt="Screenshot 2024-03-12 at 2 16 44 PM" src="https://github.com/wfxr/forgit/assets/14171107/a0a98025-0655-457c-9764-ee7152e6f795">

That is using this line:
```
get_files="echo {} | sed -e 's/^[[:space:]]*\\\\[[A-Z0-9]*\\\\][[:space:]]*//' | sed 's/ -> /\\\n/' | tr '\\\n' '\\\0'"
```


I realize my "working" version says "file added" at the top, but I'm not sure how it is supposed to look. Can you post a screenshot of what the working version looks like on your machine?

Reply:
Without my change it looks like the first one. With the change, it looks like this for me:

![image](https://github.com/wfxr/forgit/assets/88739791/d2e645f0-d79d-4f2d-8787-0ae86c982f93)

That the diff is empty is correct, since the content of the file did not change. Could you check how this behaved before #354 for you? My suggestion simply adds back the code that has been replaced in #354 by accident, so it surprises me a bit that this would break something.

Reply:
Alright, I cannot explain it but after going in and copy/pasting from the original pre-#345 version (which is identical to the code you pasted above) I can confirm this works on both Mac and Ubunutu. I pushed to this branch.

Please test the test cases again :) Thanks for the help as always